### PR TITLE
Add property to skip restoring optimizers and schedulers via plugin (DeepSpeed Checkpointing Fix 2/n) 

### PR DESCRIPTION
**Pull request by SeanNaren**
## What does this PR do?

Related to #8397

This PR adds a new property to skip optimizer/schedulers loading via the checkpoint connector, which for DeepSpeed is managed by the engine itself. The alternative was to move the loading logic into the plugin however this breaks the responsibility a bit for all other plugins that do not need this additional logic to be moved into the training type plugin.

Eventually, I'd like to see the actual loading logic end up inside the CheckpointPlugin, as described in https://github.com/PyTorchLightning/pytorch-lightning/issues/8118 to give the plugin a bit more control in the future.

CHANGELOG to be added in #8397 

cc @awaelchli @tchaton 

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/plugins/training_type/training_type_plugin.py [hunk 1]
```diff
@@ -246,6 +246,14 @@ def restore_checkpoint_after_pre_dispatch(self) -> bool:
         """
         return False
 
+    @property
+    def lightning_restore_optimizer_and_schedulers(self) -> bool:
```
Comment (justusschock) [inline pytorch_lightning/plugins/training_type/training_type_plugin.py:250]: does this really has to be a property on the instance level? wouldn't a static attribute be fine as well? I don't expect that to change per instance of the plugin, right?

**Review by justusschock — APPROVED** [review]

File: pytorch_lightning/plugins/training_type/training_type_plugin.py (hunk 1 above)
Comment (SeanNaren) [reply pytorch_lightning/plugins/training_type/training_type_plugin.py:250]: Yep definitely can be an attribute, would that be prefered?

Comment (justusschock) [reply pytorch_lightning/plugins/training_type/training_type_plugin.py:250]: From our offline sync (just for completeness): 
> No strong opinion, they both have advantages :) Class attributes are the same for all instances of the class (which they should be IMO) but properties would fail when overwriting (if not explicitly defined a setter) which also makes sense here

**Review by kaushikb11 — APPROVED** [review]
LGTM!

File: pytorch_lightning/plugins/training_type/training_type_plugin.py (hunk 1 above)
Comment (SeanNaren) [reply pytorch_lightning/plugins/training_type/training_type_plugin.py:250]: I'll keep it as is, but this was a good lesson for me as I never considered the behaviours of class attribute vs property like this before :) 

If anyone has a strong preference either way, I can change this.

**Review by tchaton — APPROVED** [review]
LGTM !
